src/main/java/sg/ncl/DataController.java

```diff
@@ -47,6 +47,8 @@
     private static final String MESSAGE_ATTRIBUTE = "message";
     private static final String EDIT_DISALLOWED = "Edit/delete of dataset disallowed as user is not contributor";
     private static final String UPLOAD_DISALLOWED = "Upload of data resource disallowed as user is not contributor";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused "UPLOAD_DISALLOWED" private field. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1068)